src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -1,771 +1,53 @@
-       * We do this to re-aggregate results from pages which are now based on single blocks,
-       * which would lead to inefficient storage performances as-is.
-       */
-      val saveDelegates = (pages: LazyPages[tezosNodeOperator.DelegateFetchingResults]) =>
-        Source
-          .fromIterator(() => pages)
-          .mapAsync(1)(identity) //extracts the future value as an element of the stream
-          .mapConcat(identity) //concatenates the list of values as single-valued elements in the stream
-          .grouped(batchingConf.blockPageSize) //re-arranges the process batching
-          .mapAsync(1)(processDelegatesPage)
-          .runFold(Monoid[Int].empty) { (processed, justDone) =>
-            processed |+| justDone
-          } andThen logOutcome
-
-      val fetchAndStore = for {
-        (delegatesPages, _) <- prunedUpdates().map(tezosNodeOperator.getDelegatesForBlocks)
-        _ <- saveDelegates(delegatesPages) flatTap cleanup
-      } yield Done
-
-      fetchAndStore.transform(
-        identity,
-        e => {
-          val error = "I failed to fetch delegates from client and update them"
-          logger.error(error, e)
-          DelegatesProcessingFailed(message = error, e)
-        }
-      )
-
-    }
-  }
-
-  /** Keeps track of time passed between different partial checkpoints of some entity processing
-    * Designed to be partially applied to set properties of the whole process once, and then only compute partial completion
-    *
-    * @param entityName a string that will be logged to identify what kind of resource is being processed
-    * @param totalToProcess how many entities there were in the first place
-    * @param processStartNanos a nano-time from jvm monotonic time, used to identify when the whole processing operation began
-    * @param processed how many entities were processed at the current checkpoint
-    */
-  private[this] def logProcessingProgress(entityName: String, totalToProcess: Int, processStartNanos: Long)(
-      processed: Int
-  ): Unit = {
-    val elapsed = System.nanoTime() - processStartNanos
-    val progress = processed.toDouble / totalToProcess
-    logger.info("================================== Progress Report ==================================")
-    logger.info("Completed processing {}% of total requested {}s", "%.2f".format(progress * 100), entityName)
+    val program = Parallel.parProduct(indexingLoop, bakerRightsSchedule).void
 
-    val etaMins = Duration(scala.math.ceil(elapsed / progress) - elapsed, NANOSECONDS).toMinutes
-    if (processed < totalToProcess && etaMins > 1) logger.info("Estimated time to finish is around {} minutes", etaMins)
-    logger.info("=====================================================================================")
+    //builds the actual resources and then pass the adapted program as a function to execute
+    useConfiguredResources(args)(program.run).as(ExitCode.Success)
```

Comment:
Notes: this combines all the prepared pieces together:
* useConfiguredResources takes the command line args to produce a set of `ConfiguredResources` (the "generic" ones) then
* passes the result to the `program.run` function, which will build the IO[Unit] value
* the `.as` will change the return value of a successful `IO` execution to the exitcode type, which is the expected return for the global `IOApp.run` function.